src/state/surface.rs

```diff
@@ -1,56 +1,65 @@
-pub async fn init_surface(window: &winit::window::Window) -> (wgpu::Surface, wgpu::Adapter) {
-    // the instance is a handle to our GPU
-    // Backends::all => Vulkan + Metal + DX12 + Browser WebGPUÒ
-    let instance = wgpu::Instance::new(wgpu::InstanceDescriptor {
-        backends: wgpu::Backends::all(),
-        dx12_shader_compiler: Default::default(),
-    });
-    let surface = unsafe { instance.create_surface(window) }.unwrap();
-    let adapter = instance
-        .request_adapter(&wgpu::RequestAdapterOptions {
-            power_preference: wgpu::PowerPreference::default(),
-            compatible_surface: Some(&surface),
-            force_fallback_adapter: false,
-        })
-        .await
-        .unwrap();
-    (surface, adapter)
-}
+use super::State;
+
+impl State {
```

Comment:
Why do the functions here need to be inside `State`? They can be free functions to be used elsewhere, can't they?

Reply:
Refactored. 
`surface.rs` is now gone and all `config` functions now go to `state/builder/init.rs`